data/modules/Scout/Scout.lua

```diff
@@ -658,7 +667,7 @@ local onShipDocked = function (player, station)
 			mission.status = "FAILED"
 		end
 
-		if station.path == mission.location and mission.status == "FAILED" or mission.status == "COMPLETED" then
+		if station.path == mission.location and (mission.status == "FAILED" or mission.status == "COMPLETED") then
```

Comment:
To double-check - should this be `mission.location` or `mission.station` now? I see you've removed an overwrite of `mission.location` in the diff above.

Reply:
Good catch, thank you. I tested that the various UI displays worked before and after completing the scanning, but I didn't test handing in the mission (it was 2am, I'd just gotten back from a 3-day motorbiking weekend a few hours earlier..)

Reply:
Fixed (and tested) in 8d6092a78ec91a503274e0e6c5f12395a6a45cf1

Tested both landing at an incorrect station after completing a scan, as well as at the correct station, following setting a new destination on scan completion.